### DISCUSSION TITLE
Ability to catch errors in swr hook onError and global SWRConfig onError at the same time

### DISCUSSION THREAD

**Discussion by nik-webdevelop** [vercel/swr#2556] [Q&A] [answered]
Is there a way to catch errors on swr hook error handler and global SWRConfig error handler at the same time?
I need to log errors/show toasts globally and at the same handle some of the errors locally.

Currently, if I set the onError in the hook's config, the global onError is not called.

[CodeSandbox](https://codesandbox.io/s/infallible-easley-e93myo?file=/src/App.tsx)

SWR version. 
2.1.1

**Comment by koba04** [comment, marked as answer]
You can use the global onError by calling it from the onError in the component.

```jsx
const Component = () => {
  const { onError } = useSWRConfig();
  useSWR("/api/data", (url) => fetch(url).then((r) => r.json()), {
    onError: (...args) => {
      const [error] =args;
      console.log("hook onError:", error);
      onError(...args);
    }
  });

  return null;
};
```

**Reply by nik-webdevelop** [reply]
Great, thanks a lot!
